src/common/core/coretype.js

```diff
@@ -177,117 +177,110 @@ define([
             return node;
         }
 
+        /**
```

Comment:
If these are often called in conjunction, we could consider having a combined method (to me it looks like they are doing the same iterations).